homeassistant/components/amcrest/__init__.py

```diff
@@ -105,6 +115,10 @@ def _has_unique_names(devices):
                     _deprecated_sensor_values),
         vol.Optional(CONF_SWITCHES):
             vol.All(cv.ensure_list, [vol.In(SWITCHES)]),
+        vol.Optional(CONF_MAX_ERRORS, default=DEFAULT_MAX_ERRORS):
```

Comment:
I don't think that these should be configurable. We should pick some decent defaults that work for the device and stick with it.

Reply:
Done.

Unfortunately, depending on configuration, the rate at which errors can happen can vary widely (i.e., with more sensors configured, services called, etc, there will be more frequent communication.) It might make sense to measure the time period over which errors continue to happen and if it becomes too long, _then_ make the device unavailable. We'll go with this for now and I'll monitor how it goes. Might need to switch to time based error monitoring instead of count based in a follow-up PR.